### ISSUE TITLE
Problem importing Flight Demo (2023 Branch)

### ISSUE THREAD

**Issue by mydoghasfleas** [abapGit/abapGit#6927]
On our on-prem S/4 HANA 2023 system, I am importing the 2023 branch of the flight model demo (https://github.com/SAP-samples/abap-platform-refscen-flight) from an exported zip file, as our system is blocked by the firewall from accessing github.com.

This time, instead of using the version of the standalone abapGit program shipped with our S/4 HANA system, I have downloaded and installed the latest (1.128.0) and used that.

After apparently successfully importing and activating all objects, I still am missing some, like the number range objects contained in the package (e.g. /DMO/AGNCY).

I am experiencing the same issue as reported here: https://github.com/SAP-samples/abap-platform-refscen-flight/issues/26.

I can confirm that I see the number range objects listed when I pull from the local repository (zip), after checking they are in the repository too.

So I decide to pull again and am surprised that there are many other objects listed that would apparently be created (again):

![image](https://github.com/abapGit/abapGit/assets/125179869/0734356d-c75a-414e-8727-18d1dd1b42b6)

Including the NROs:

![image](https://github.com/abapGit/abapGit/assets/125179869/e8badba3-de87-4531-82ba-721e88e63b5c)

To be safe, I first deselect everything and try (re)import only the NROs.

Again, this is apparently successful, but still the NROs are not created in our system.

![image](https://github.com/abapGit/abapGit/assets/125179869/c70bdceb-6273-4a2a-b232-374abf23a46b)

(I'm pretty sure I saw it try deserialize someting other than those objects too, but I can't say 100% for sure).

I will not go further at this point as perhaps you can advise me on the best course of action, before I break something.

**Comment by mydoghasfleas** [comment]
Strangely, I now see some objects not yet activated.

![image](https://github.com/abapGit/abapGit/assets/125179869/7def75a8-aca9-418c-9de7-c3e840daef2d)

I choose Advanced -> Activate Objects, like before, and this time get a popup with a whole bunch of objects to activate.

![image](https://github.com/abapGit/abapGit/assets/125179869/7ca6cd7b-617b-4968-9a87-f68ab3f206e6)

Then I get a whole bunch of errors:

![image](https://github.com/abapGit/abapGit/assets/125179869/5d5bf583-fac1-4220-a03d-ea70d59e5788)

I think I will exit the program first to release those locks. Clicking back brings up the activation dialog again. I continue, and this time itself reports the errors:

![image](https://github.com/abapGit/abapGit/assets/125179869/3f52851a-be90-4e33-ae12-46e02efeedc0)

So I cancel, and at this point, it is still doing something, I do not know what:

![image](https://github.com/abapGit/abapGit/assets/125179869/dfe0c277-1a55-48f0-a7c6-794dfe98c2e0)

**Comment by mydoghasfleas** [comment]
OK, so after getting out, I went back in to try and load the repo from the zip again.

By the way, when I created the offline repo this time, the name was somehow not adopted:

![image](https://github.com/abapGit/abapGit/assets/125179869/705b196f-1823-4f11-a2d5-b0d78aa69aca)

Now I am ready to pull again from the zip, but is it safe to try import all these objects?

![image](https://github.com/abapGit/abapGit/assets/125179869/2c3ba75d-b653-42b3-ab99-99fcc52d176f)

![image](https://github.com/abapGit/abapGit/assets/125179869/61a2c13b-c617-45d1-b31d-7224b2f48692)

This time, I throw caution to the wind and proceed with these objects. (I don't know what object type DDLX or SRVB is, but I do not see it in the dropdown in the TADIR selection either).

The question is of course, why did they not get pulled the first time?

Now I try activate again, but it stops (no object list dialog this time, by the way):

![image](https://github.com/abapGit/abapGit/assets/125179869/59ae22c3-8369-450c-9439-1be5aa24e6c6)

**Comment by mydoghasfleas** [comment]
Sorry, after documenting this here, I remembered that this should probably go to SAP-samples/abap-platform-refscen-flight, so I have logged it there, referencing this one: https://github.com/SAP-samples/abap-platform-refscen-flight/issues/36

**Comment by mbtools** [comment]
Please test with https://github.com/abapGit-tests/NROB to see if importing number ranges works in general.

**Comment by mydoghasfleas** [comment]
> Please test with https://github.com/abapGit-tests/NROB to see if importing number ranges works in general.

@mbtools So just to confirm: How do I use this? Is this a plugin for abapGit? Do I import this with abapGit and then is the functionality available?

**Comment by mbtools** [comment]
It's just a test repo. Pull it into your system with abapGit. Does it create the number range object correctly?

**Comment by mydoghasfleas** [comment]
Ah, understood, thanks @mbtools.

I tested it and it successfully pulled the NRO from a zip file and created it on the system:

![image](https://github.com/abapGit/abapGit/assets/125179869/bc60b7c9-c108-49b2-8b52-8c7b7a5f83be)

**Comment by mydoghasfleas** [comment]
@mbtools Can any of the settings, when creating the repository, have anything to do with the problems I am experiencing?

![image](https://github.com/abapGit/abapGit/assets/125179869/87df54eb-cbfc-40dc-93f3-2a40e5848740)

**Comment by SugguSandeep29** [comment]
> @mbtools Can any of the settings, when creating the repository, have anything to do with the problems I am experiencing?
> 
> ![image](https://private-user-images.githubusercontent.com/125179869/330685470-87df54eb-cbfc-40dc-93f3-2a40e5848740.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3NjM3MzgsIm5iZiI6MTcxNTc2MzQzOCwicGF0aCI6Ii8xMjUxNzk4NjkvMzMwNjg1NDcwLTg3ZGY1NGViLWNiZmMtNDBkYy05M2YzLTJhNDBlNTg0ODc0MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNVQwODU3MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NGI3ZjhkNzcwOTBhYWVjOTc0OTY1ZmE0NDU2MTRkNjc5NzNhNzI0OTY4ZDI3ZmFjNDJmNDlkNTYyZjY0N2Y4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PrIiF3Sa1meixr3YMJds_Rcce2Yvu48S6AyWtcOn3gU)

Hi @mydoghasfleas 

That's not a problem, it's your choice 

Soon after bringing the package on abapGit Platform, Operation Need to perform (Advanced > Activate Objects)
Once it's done need to perform (Push or Pull) Offline/Online 

Thanks, 
Suggu Sandeep.

**Comment by mbtools** [comment]
Closing this in favor of https://github.com/SAP-samples/abap-platform-refscen-flight/issues/36
